application/server/server_smoke_test.go

```diff
@@ -605,23 +611,67 @@ func checkForScanParams(t *testing.T, jsonRPCRecorder *testutil.JsonRPCRecorder,
 		}
 		return false
 	}, 10*time.Second, 10*time.Millisecond)
-	return scanParams
 }
 
-func checkAutofixDiffs(t *testing.T, c *config.Config, snykCodeScanParams types.SnykScanParams, loc server.Local) {
+func getIssueListFromPublishDiagnosticsNotification(t *testing.T, jsonRPCRecorder *testutil.JsonRPCRecorder, p product.Product, folderPath string) []types.ScanIssue {
+	t.Helper()
+
+	var issueList []types.ScanIssue
+	notifications := jsonRPCRecorder.FindNotificationsByMethod("textDocument/publishDiagnostics")
+	for _, n := range notifications {
+		diagnosticsParams := types.PublishDiagnosticsParams{}
+		_ = n.UnmarshalParams(&diagnosticsParams)
+		for _, diagnostic := range diagnosticsParams.Diagnostics {
+			diagnosticCode, ok := diagnostic.Code.(string)
+			if ok && diagnosticCode == "Snyk Error" {
+				continue
+			}
+			if diagnostic.Source != string(p) || filepath.Dir(uri.PathFromUri(diagnosticsParams.URI)) != folderPath {
+				continue
+			}
+
+			var issue types.ScanIssue
+			err := mapSerializedIssueToStruct(diagnostic.Data, &issue)
+
+			if err != nil {
+				continue
+			}
+			issueList = append(issueList, issue)
+		}
+	}
+
+	return issueList
+}
+
+func mapSerializedIssueToStruct(m any, result *types.ScanIssue) error {
+	// Serialize the map to JSON
+	serializedMap, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+
+	// Deserialize the JSON back to the struct
+	if err = json.Unmarshal(serializedMap, result); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkAutofixDiffs(t *testing.T, c *config.Config, issueList []types.ScanIssue, loc server.Local) {
 	t.Helper()
 	if isNotStandardRegion(c) {
 		return
 	}
-	assert.Greater(t, len(snykCodeScanParams.Issues), 0)
-	for _, issue := range snykCodeScanParams.Issues {
+	assert.Greater(t, len(issueList), 0)
+	for _, issue := range issueList {
 		codeIssueData, ok := issue.AdditionalData.(map[string]interface{})
 		if !ok || codeIssueData["hasAIFix"] == false || codeIssueData["rule"] != "WebCookieSecureDisabledByDefault" {
 			continue
 		}
 		call, err := loc.Client.Call(ctx, "workspace/executeCommand", sglsp.ExecuteCommandParams{
 			Command:   types.CodeFixDiffsCommand,
-			Arguments: []any{uri.PathToUri(snykCodeScanParams.FolderPath), uri.PathToUri(issue.FilePath), issue.Id},
+			Arguments: []any{path.Base(issue.FilePath), uri.PathToUri(issue.FilePath), issue.Id},
```

Comment:
This doesn't look correct. It should be folderPath, filePath. I'd pass in the folder or folderPath from the caller.

Reply:
Agree, it doesn't make any sense lol